WowPacketParser/SQL/QueryBuilder.cs

```diff
@@ -90,6 +99,97 @@ public string Build()
+
+                    if (value == null ||
+                        (_onlyPrimaryKeys &&
+                         field.Item3.Any(a => !a.IsPrimaryKey)))
+                        continue;
+
+                    whereCondition.Add(field.Item1, SQLUtil.ToSQLValue(value));
+                }
+
+                if (!whereCondition.IsEmpty)
+                    whereConditions.Add(whereCondition);
+            }
+
+            // 2. Check if all conditions share the same fields
+            bool allShareSameFields = true;
+            var baselineConditionFields = whereConditions.First().FieldValuePairs.Keys;
+            foreach (var condition in whereConditions)
+                if (!condition.FieldValuePairs.Keys.SequenceEqual(baselineConditionFields))
+                    allShareSameFields = false;
+
+            if (!allShareSameFields)
+                return null;
+
+            // ToDo: support the case with more than 2 fields
+            if (baselineConditionFields.Count != 2)
+                return null;
+
+            // 3. Get a distinct of values for each field
+            var fieldsWithDistinctValues = (from condition in whereConditions
+                                            from fieldValuePair in condition.FieldValuePairs
+                                            group fieldValuePair by fieldValuePair.Key into byField
+                                            select new { Field = byField.Key, Values = byField.ToList().Select(pair => pair.Value).Distinct().ToList() }
+                                            ).OrderBy(field => field.Values.Count()).ToList();
+
+            // 4. Get the field with lowest different values
+            var fieldWithLowestDifferentValues = fieldsWithDistinctValues.First().Field;
+            // ToDo: support the case with more than 2 fields
+            var secondField = fieldsWithDistinctValues.Skip(1).First().Field;
+
+            // 5. Group conditions by the field with lowest different values
+            var conditionsByFieldWithLowestDifferentValues = whereConditions.GroupBy(cond => cond.FieldValuePairs[fieldWithLowestDifferentValues])
+                                                            .ToDictionary(g => g.Key, g => g.ToList());
+
+            // 6. Build the where clause using the field with lowest different values for the first condition
+            StringBuilder whereClause = new StringBuilder();
+            foreach(var pair in conditionsByFieldWithLowestDifferentValues)
+            {
+                whereClause.Append("(");
+                whereClause.Append(fieldWithLowestDifferentValues);
+                whereClause.Append("=");
+                whereClause.Append(pair.Key);
+                whereClause.Append(" AND ");
+
+                whereClause.Append(secondField);
+                whereClause.Append(" IN (");
+                whereClause.Append(string.Join(',', pair.Value.Select(cond => cond.FieldValuePairs[secondField])));
```

Comment:
if pair.Value.Select(cond => cond.FieldValuePairs[secondField]) has Length 1, instead of field IN (1) we could do field = 1

Reply:
yep